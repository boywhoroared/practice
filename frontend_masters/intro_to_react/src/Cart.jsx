import currencyFormatter from "./currencyFormatter.js";

const intl = currencyFormatter;

export default function Cart({ cart, checkout }) {
  // the `checkout` prop will be the checkout fn passed down from the Cart

  let total = 0;

  // Not written as a reduce because it's more understandable
  // (less smug satisfaction)
  for (let i = 0; i < cart.length; i++) {
    const current = cart[i];
    total += current.pizza.sizes[current.size];
  }

  return (
    <div className="cart">
      <h2>Cart</h2>
      <ul>
        {cart.map((item, index) => (
          <li key={index}>
            <span className="size">{item.size}</span> –&nbsp;
            <span className="type">{item.pizza.name}</span> –&nbsp;
            <span className="price">{item.price}</span>
          </li>
        ))}
      </ul>
      <p>Total: {intl.format(total)}</p>
      <button onClick={checkout}>Checkout</button>
    </div>
  );
}
