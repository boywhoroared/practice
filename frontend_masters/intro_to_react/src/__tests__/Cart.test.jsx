import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Cart from "../Cart";

test("snapshot with nothing in cart", () => {
  const { asFragment } = render(<Cart cart={[]} />);

  // The first time this test is run, it will create a snapshot.

  // Subsequent runs will compare the rendered fragment to the contents of the
  // snapshot

  // You will want to commit the snapshots in your repo so the snapshots are
  // present and changes can be tracked.
  expect(asFragment()).toMatchSnapshot();

  // After running this test once, go to `Cart.jsx`
  // and change the heading from `Cart` to `Fart`
  // When test runs again, it should fail.

  // If this change is what you intended, and not
  // actually a failure, you can press 'u' to update
  // the snapshot (then commit it to your repo).
});

test("API snaphot", () => {
  // Response from backend
  const pizzaOfTheDayAPIResponse = {
    id: "calabrese",
    name: "The Calabrese Pizza",
    category: "Supreme",
    description: "Pizza from Calabria",
    image: "/public/pizzas/calabrese.webp",
    size: { S: 12.25, M: 16.25, L: 20.25 },
  };

  // Assume that `pizzaOfTheDayAPIRespose` is the result of running your API
  // library code (might be a mock? or staging service with fake data, or against a real service)
  //
  // This functions like an acceptance or end-to-end test where we can see if the
  // API contracts are satisfied/broken. Can help keeping FE and BE in sync.
  expect(pizzaOfTheDayAPIResponse).toMatchSnapshot();
});
