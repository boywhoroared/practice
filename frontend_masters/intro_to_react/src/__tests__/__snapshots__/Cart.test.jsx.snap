// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`API snaphot 1`] = `
{
  "category": "Supreme",
  "description": "Pizza from Calabria",
  "id": "calabrese",
  "image": "/public/pizzas/calabrese.webp",
  "name": "The Calabrese Pizza",
  "size": {
    "L": 20.25,
    "M": 16.25,
    "S": 12.25,
  },
}
`;

exports[`snapshot with nothing in cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Fart
    </h2>
    <ul />
    <p>
      Total: BBDÂ 0.00
    </p>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;
